File: scratch/check_eval_results.ts
import fs from "fs";
import path from "path";

const envPath = path.resolve(__dirname, "../.env");
if (fs.existsSync(envPath)) {
    const envContent = fs.readFileSync(envPath, "utf8");
    const lines = envContent.split("\n");
    for (const line of lines) {
        if (line.startsWith("DATABASE_URL=")) {
            const val = line.substring("DATABASE_URL=".length).trim().replace(/"/g, "");
            process.env.DATABASE_URL = val;
            break;
        }
    }
}

async function main() {
    const { prisma } = await import("../src/lib/db");
    const rows = await prisma.preRevision.findMany({
        include: {
            entrega: {
                include: {
                    escuela: true,
                    periodoEntrega: {
                        include: { programa: true }
                    }
                }
            }
        },
        orderBy: { updatedAt: "desc" }
    });

    console.log(`Total PreRevisions in DB: ${rows.length}`);
    for (const r of rows) {
        const res = r.resultado as any;
        let status = "N/A";
        if (res) {
            status = res.explicacion || "N/A";
            if (status.includes("Error") || status.includes("fallaron")) {
                status = "❌ ERROR: " + status.substring(0, 100);
            } else {
                status = "✅ SUCCESS: " + status.substring(0, 100);
            }
        }
        console.log(`Escuela: ${r.entrega.escuela.nombre}, Programa: ${r.entrega.periodoEntrega.programa.nombre}, Updated: ${r.updatedAt.toISOString()}, Status: ${status}`);
    }
}

main().catch(console.error);

File: scratch/check_eval_results.py
import os
import psycopg2
from psycopg2.extras import RealDictCursor

# Find DATABASE_URL
env_path = os.path.join(os.path.dirname(__file__), "../.env")
db_url = None
if os.path.exists(env_path):
    with open(env_path, "r", encoding="utf-8") as f:
        for line in f:
            if line.startswith("DATABASE_URL="):
                db_url = line.split("=")[1].strip().strip('"').strip("'")
                break

if not db_url:
    print("DATABASE_URL not found")
    exit(1)

conn = psycopg2.connect(db_url)
cur = conn.cursor(cursor_factory=RealDictCursor)

cur.execute("""
    SELECT pr.id, pr."entregaId", pr."updatedAt", esc.nombre as escuela, prog.nombre as programa, pr.resultado
    FROM "PreRevision" pr
    JOIN "Entrega" ent ON pr."entregaId" = ent.id
    JOIN "Escuela" esc ON ent."escuelaId" = esc.id
    JOIN "PeriodoEntrega" pe ON ent."periodoEntregaId" = pe.id
    JOIN "Programa" prog ON pe."programaId" = prog.id
    ORDER BY pr."updatedAt" DESC
""")

rows = cur.fetchall()
print(f"Total PreRevisions in DB: {len(rows)}")
for r in rows:
    res = r["resultado"]
    status = "N/A"
    if res:
        if isinstance(res, dict):
            status = res.get("explicacion", "N/A")
            if "Error" in status or "fallaron" in status:
                status = "❌ ERROR: " + status[:100]
            else:
                status = "✅ SUCCESS: " + status[:100]
        else:
            status = str(res)[:100]
    print(f"Escuela: {r['escuela']}, Programa: {r['programa']}, Updated: {r['updatedAt']}, Status: {status}")

cur.close()
conn.close()
